src/lib/stringifiers.ts

```diff
@@ -0,0 +1,151 @@
+import * as t from './models/types';
```

Comment:
So this isn't your fault, it was a problem with the existing code but nobody ran into it since it wasn't used externally... But if we are going to more publicly expose this functionality we need to fix this. The serializers can generate incorrect types. Consider this:

```
const broken = new t.IntersectionType([
  new t.ReferenceType('T', t.ReferenceType.SYMBOL_FQN_RESOLVED),
  new t.UnionType([
    new t.ReferenceType('U', t.ReferenceType.SYMBOL_FQN_RESOLVED),
    new t.ReferenceType('V', t.ReferenceType.SYMBOL_FQN_RESOLVED)
  ])
])
```

This will currently stringify to `T & U | V` while the actual type is `T & (U | V)`

There's also a problem with function types and conditional types. `() => any extends 1 ? 1 : 2` is different than `(() => any) extends 1 ? 1 : 2`.

Reply:
Agreed, I did notice there could be some precedence issues with complex union/intersection types. We'll have to add some parenthesizing logic.